test/integration/src/test/java/io/pravega/test/integration/ReaderGroupNotificationTest.java

```diff
@@ -251,4 +253,64 @@ public void testEndOfStreamNotifications() throws Exception {
         assertTrue("Listener invoked", listenerInvoked.get());
     }
 
+    @Test(timeout = 40000)
+    public void testSegmentTruncatedNotifications() throws Exception {
+        final String streamName = "stream2";
+        StreamConfiguration config = StreamConfiguration.builder()
+                                                        .scope(SCOPE)
+                                                        .streamName(streamName)
+                                                        .scalingPolicy(ScalingPolicy.byEventRate(10, 2, 1))
+                                                        .build();
+        Controller controller = controllerWrapper.getController();
+        controllerWrapper.getControllerService().createScope(SCOPE).get();
+        controller.createStream(config).get();
+        @Cleanup
+        ConnectionFactory connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder()
+                                                                                    .controllerURI(URI.create("tcp://localhost"))
+                                                                                    .build());
+        @Cleanup
+        ClientFactory clientFactory = new ClientFactoryImpl(SCOPE, controller, connectionFactory);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+        writer.writeEvent("0", "data1").get();
+
+        // truncate
+
+        @Cleanup
+        ReaderGroupManager groupManager = new ReaderGroupManagerImpl(SCOPE, controller, clientFactory,
+                connectionFactory);
+        groupManager.createReaderGroup("reader", ReaderGroupConfig
+                .builder().disableAutomaticCheckpoints().stream(Stream.of(SCOPE, streamName)).build());
+        @Cleanup
+        ReaderGroup readerGroup = groupManager.getReaderGroup("reader");
+        @Cleanup
+        EventStreamReader<String> reader1 = clientFactory.createReader("readerId", "reader", new JavaSerializer<>(),
+                ReaderConfig.builder().build());
+        Checkpoint cp = readerGroup.initiateCheckpoint("myCheckpoint", executor).join();
+        StreamCut streamCut = cp.asImpl().getPositions().values().iterator().next();
+        Boolean result = controller.truncateStream(SCOPE, streamName, streamCut).get();
```

Comment:
This seems to be trying to test the high level behavior. Could we instead have a unit test that simply checks that delegation token is properly refreshed upon finding a truncated segment?